libsolidity/parsing/Parser.cpp

```diff
@@ -94,14 +94,22 @@ ASTPointer<SourceUnit> Parser::parse(CharStream& _charStream)
 		m_recursionDepth = 0;
 		m_scanner = make_shared<Scanner>(_charStream);
 		ASTNodeFactory nodeFactory(*this);
+		bool finishedParsingPragmas = false;
+		m_experimentalSolidityEnabledInCurrentSourceUnit = false;
```

Comment:
Why not just the below? Does we really need to be explicit about it been enabled in the current source unit? I mean, if it is enable, it should be for all sources, right?

```suggestion
		m_experimentalSolidityEnabled = false;
```

Reply:
Nope, since we do parsing per source unit, meaning that each source unit should use the default parsing behaviour if the pragma has not been declared, otherwise the experimental parsing. The actual error for not having the experimental pragma declared in every source unit happens in the analysis stage, so after parsing, meaning that the parsing should succeed if the actual code conforms to the parsing behaviour (dictated by the experimental pragma), and only fail later. But in any case, [Daniel already asked for the per source unit one to remain](https://github.com/ethereum/solidity/pull/14168#discussion_r1182573874) :D